app/src/main/java/mozilla/lockbox/presenter/ItemListPresenter.kt

```diff
@@ -9,38 +9,61 @@ package mozilla.lockbox.presenter
 import android.support.annotation.IdRes
 import io.reactivex.Observable
 import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.rxkotlin.Observables
 import io.reactivex.rxkotlin.addTo
 import mozilla.lockbox.R
 import mozilla.lockbox.action.DataStoreAction
+import mozilla.lockbox.action.ItemListSort
 import mozilla.lockbox.action.RouteAction
+import mozilla.lockbox.action.SettingsAction
 import mozilla.lockbox.extensions.mapToItemViewModelList
 import mozilla.lockbox.flux.Dispatcher
 
 import mozilla.lockbox.flux.Presenter
 import mozilla.lockbox.log
 import mozilla.lockbox.model.ItemViewModel
 import mozilla.lockbox.store.DataStore
+import mozilla.lockbox.store.PreferencesStore
 
 interface ItemListView {
     val itemSelection: Observable<ItemViewModel>
     val filterClicks: Observable<Unit>
     val menuItemSelections: Observable<Int>
+    val sortItemSelection: Observable<ItemListSort>
     fun updateItems(itemList: List<ItemViewModel>)
+    fun updateItemListSort(sort: ItemListSort)
 }
 
 class ItemListPresenter(
     private val view: ItemListView,
     private val dispatcher: Dispatcher = Dispatcher.shared,
-    private val dataStore: DataStore = DataStore.shared
+    private val dataStore: DataStore = DataStore.shared,
+    private val prefsStore: PreferencesStore = PreferencesStore.shared
 ) : Presenter() {
+
     override fun onViewReady() {
-        dataStore.list
+        val itemViewModelList = dataStore.list
                 .filter { it.isNotEmpty() }
                 .mapToItemViewModelList()
+
+        Observables.combineLatest(itemViewModelList, prefsStore.itemListSortObservable)
+                .distinctUntilChanged()
+                .map { pair ->
+                    when (pair.second) {
+                        ItemListSort.ALPHABETICALLY -> { pair.first.sortedBy { it.title } }
+                        ItemListSort.RECENTLY_USED -> { pair.first.sortedBy { -it.timeLastUsed } }
```

Comment:
Done as in done.

Reply:
Hmm. I've moved this over, because I think you're right, it shouldn't be a done on the view model. 

However, I'm concerned that we are calling `titleFromHostname` way to many times.